Realm/RLMArrayLinkView.mm

```diff
@@ -138,7 +141,7 @@ - (void)addObject:(RLMObject *)object {
     RLMLinkViewArrayValidateInWriteTransaction(self);
 
     RLMValidateObjectClass(object, self.objectClassName);
-    if (object.realm != self.realm) {
+    if (object->_realm != self.realm) {
```

Comment:
I think using `_realm` in all the spots where we're comparing it to `object->_realm` would look a little better. Mixing ivar and property syntax for the same sort of thing looks a little off.